packages/types/src/index.ts

```diff
@@ -267,6 +267,25 @@ export interface Timepicker extends Action {
   timezone?: string;
 }
 
+/**
+ * @description An element that allows the selection of a time of day formatted as a UNIX timestamp. On desktop
+ * clients, this time picker will take the form of a dropdown list and the date picker will take the form of a dropdown
+ * calendar. Both options will have free-text entry for precise choices. On mobile clients, the time picker and date
+ * picker will use native UIs.
+ * {@link https://api.slack.com/reference/block-kit/block-elements#datetimepicker}
+ */
+export interface DateTimepicker extends Action {
+  type: 'datetimepicker';
+  /**
+   * @description The initial date and time that is selected when the element is loaded, represented as a UNIX
+   * timestamp in seconds. This should be in the format of 10 digits, for example 1628633820 represents the date and
+   * time August 10th, 2021 at 03:17pm PST.
+   */
+  initial_date_time?: number;
+  confirm?: Confirm;
```

Comment:
FYI I didn't document `confirm`, `placeholder`, `focus_on_load` and other common properties across elements as I plan to send a PR to factor these out into dedicated interfaces that the other elements can then extend from (should DRY up the types a little and also make it easier to document the properties in one place).